### ISSUE TITLE
Web - Endless loading on Private section when login via Abracadabra

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#36647]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.42-0
**Reproducible in staging?:** Y
**Reproducible in production?:** N
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):** gocemate@gmail.com
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

## Action Performed:
1. Navigate to staging.new.expensify.com
2. Enter gmail account with lot of conversations> Click Continue
3. Navigate to e-mail> copy the login link, change to staging and Enter in new tab
4. Navigate to original Tab and while app still loading navigate to Settings icon
5. Take a note of the Private section

## Expected Result:
Private section should load immediately or there should be short time delay

## Actual Result:
There is a endless loading on Private section

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/ba1ef566-5888-464f-9b5b-ddb4737ae44b

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by lanitochka17** [comment]
Production: 

https://github.com/Expensify/App/assets/78819774/d74849f6-44e7-4421-8f33-d1c46be4646f

**Comment by NikkiWines** [comment]
Yeah, weird - looks like `private_personalDetails` is getting set to `[]` on staging. I can't reproduce on dev, and I don't see any logic. The details are being returned correctly in `OpenPersonalDetailsPage` though. 

FWIW this bug also occurs if you manually paste the magic code, so it's not specific to the `Abracadabra` flow.

**Comment by NikkiWines** [comment]
@grgia could this be tied to https://github.com/Expensify/Web-Expensify/pull/40867 by any chance?

**Comment by grgia** [comment]
@NikkiWines it looks like personalDetails arent loading here, not privatePersonalDetails

**Comment by NikkiWines** [comment]
@grgia the personal details seem to be ok from my testing - it looks like only the private ones are getting overwritten and in the response to the `OpenApp` call we can see the following: 

![image](https://github.com/Expensify/App/assets/16747822/43ec210e-2270-4394-8a30-b6538abdfc87)

Even though the private_personalDetails are correctly being returned earlier in `OpenPersonalDetailsPage`

![image](https://github.com/Expensify/App/assets/16747822/9574a706-7446-4bd9-8f45-8a30bafdde41)

It also explains why this bug only occurs if you immediately navigate to the settings page (prompting the `OpenPersonalDetailsPage`, since the `OpenApp` call takes a lot longer and can finish after `OpenPersonalDetailsPage` completes.

**Comment by grgia** [comment]
ah gotcha, so we need to remove the private personal details from being overwritten in `OpenPersonalDetailsPage`

**Comment by NikkiWines** [comment]
Yeah, I think ideally `OpenApp` would return the default empty values ([here](https://github.com/Expensify/App/blob/9843f6641f65d11e125c4325df1f95feae9d1259/src/types/onyx/PrivatePersonalDetails.ts#L9-L20)) for `private_personalDetails` instead of `[]` [here](https://github.com/Expensify/Web-Expensify/pull/40867/files#diff-0df2c659854969ada0990bff0abf41c3d4afc78662764aa968c4e2c61e29a823R167).

**Comment by NikkiWines** [comment]
Oh also not a newDot blocker since this is due to backend changes 👍

**Comment by damadoko** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Endless loading on personal private section when logging

### What is the root cause of that problem?
<img width="812" alt="Screenshot 2024-02-16 at 17 59 16" src="https://github.com/Expensify/App/assets/57703073/7401044c-c2e7-4a6b-b5f2-19e92eba00d2">.

1. When user click fast enough after login via Abracadabra, the `OpenPersonalDetailsPage` was called before `OpenApp`. 
2. `OpenPersonalDetailsPage` return appropriate `private_personalDetails` data.
3. `OpenApp` return `[]` for `private_personalDetails` data (and override the appropriate one).
4. Personal private section keep loading because below line fallback to `true`.

https://github.com/Expensify/App/blob/a6fe014150f0de5ddf86787b05fd3417447c73f5/src/pages/settings/Profile/ProfilePage.js#L106

### What changes do you think we should make in order to solve the problem?
Currently, user can navigate to setting page even when the user's avatar icon is show loading.
=> We need to prevent navigate action when loading to ensure `OpenPersonalDetailsPage` can only be called after `OpenApp`

The condition should be apply in line 44:
https://github.com/Expensify/App/blob/a6fe014150f0de5ddf86787b05fd3417447c73f5/src/pages/home/sidebar/PressableAvatarWithIndicator.js#L43-L50

It should be
```js
const shouldShowLoading = isLoading && !currentUserPersonalDetails.avatar;
const showSettingsPage = useCallback(() => {
  if (isCreateMenuOpen || shouldShowLoading) {
    return;
  }
```

### What alternative solutions did you explore? (Optional)
Above solution correct behavior of loading element. Anyway, we can change `OpenApp` in the BE too. Both fix the issue

**Comment by grgia** [comment]
I can take this @NikkiWines

**Comment by grgia** [comment]
also the auth changes weren't merged yet oops https://github.com/Expensify/Auth/pull/9913

**Comment by s77rt** [comment]
@damadoko Thank you for your interest here. Looks this will be fixed internally.

**Comment by NikkiWines** [comment]
Re-assigning to @grgia per [this comment](https://github.com/Expensify/App/issues/36647#issuecomment-1949202241)

**Comment by joekaufmanexpensify** [comment]
Only payment needed here is $500 to @s77rt for C+ via Upwork (this issue was created before default price lowered to $250).

**Comment by joekaufmanexpensify** [comment]
OG upwork job is closed. new one is here: https://www.upwork.com/jobs/~01058cf92b6ef09600

**Comment by joekaufmanexpensify** [comment]
@s77rt offer for $500 sent!

**Comment by s77rt** [comment]
@joekaufmanexpensify Accepted! Thanks!

**Comment by joekaufmanexpensify** [comment]
@s77rt $500 sent and contract ended!

**Comment by joekaufmanexpensify** [comment]
Upwork job closed.
